### PR TITLE
Fix defaultValue props update on autocomplete component

### PR DESCRIPTION
**Pull request by grosbouddha**

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -3364,7 +3364,7 @@
     },
     "ansi-escapes": {
       "version": "1.4.0",
-      "resolved": "https://registry.npmjs.org/ansi-escapes/-/ansi-escapes-1.4.0.tgz",
+      "resolved": "http://registry.npmjs.org/ansi-escapes/-/ansi-escapes-1.4.0.tgz",
```
Comment (pith) [inline package-lock.json]: As you didn't change the `package.json` the `package-lock.json` should be updated. You can use `npm install --no-save` to update your deps without updating the package lock.

File: src/autoComplete/index.tsx [hunk 2]
```diff
@@ -168,19 +192,24 @@ export default class AutoComplete extends Component<AutoCompleteProps, AutoCompl
   onInputChange = ({ value }: { value: query }) => {
     this.currentValue = value
     if (this.hasMinCharsForSearch()) {
-      this.setState({ noResults: false, query: value }, this.searchForItems)
+      this.setState({
+        textfieldValue: String(value),
+        noResults: false, query: value },
```
Comment (pith) [inline src/autoComplete/index.tsx]: I'm surprised that prettier didn't change this indentation

File: src/autoComplete/index.tsx [hunk 3]
```diff
@@ -168,19 +192,24 @@ export default class AutoComplete extends Component<AutoCompleteProps, AutoCompl
   onInputChange = ({ value }: { value: query }) => {
     this.currentValue = value
     if (this.hasMinCharsForSearch()) {
-      this.setState({ noResults: false, query: value }, this.searchForItems)
+      this.setState({
+        textfieldValue: String(value),
```
Comment (pith) [inline src/autoComplete/index.tsx]: It's a good idea to separate the textfieldValue from the query as they have different concerns. Can you check however if we can remove `this.currentValue` it seems to be just a copy of the query.

File: src/autoComplete/index.tsx [hunk 4]
```diff
@@ -128,8 +134,23 @@ export default class AutoComplete extends Component<AutoCompleteProps, AutoCompl
...
 
   static getDerivedStateFromProps(props: AutoCompleteProps, state: AutoCompleteState) {
+    const hasDefaultValueChanged = state.lastDefaultValue !== props.defaultValue
+    const baseState = hasDefaultValueChanged ? {
+      textfieldValue: props.defaultValue,
+      lastDefaultValue: props.defaultValue,
+      items: [],
+    } : {}
+
+    if (state.lastDefaultValue !== props.defaultValue) {
```
Comment (loick) [inline src/autoComplete/index.tsx]: `if(hasDefaultValueChanged)`

File: src/autoComplete/index.tsx (hunk 2 above)
Comment (loick) [reply src/autoComplete/index.tsx]: +1

Comment (loick) [reply src/autoComplete/index.tsx]: try to run `npm run prettier`, just in case?

File: src/autoComplete/index.tsx [hunk 5]
```diff
@@ -248,7 +278,7 @@ export default class AutoComplete extends Component<AutoCompleteProps, AutoCompl
           onBlur={this.props.onBlur}
           onClear={this.props.onClear}
           placeholder={this.props.placeholder}
-          defaultValue={String(this.state.query)}
+          defaultValue={String(this.state.textfieldValue)}
```
Comment (loick) [inline src/autoComplete/index.tsx]: this.state.textfieldValue is already a string

File: src/autoComplete/index.tsx (hunk 4 above)
Comment (grosbouddha) [reply src/autoComplete/index.tsx]: Done

File: src/autoComplete/index.tsx (hunk 2 above)
Comment (grosbouddha) [reply src/autoComplete/index.tsx]: npm run prettier did help.
Is it called automatically? when?

File: src/autoComplete/index.tsx (hunk 3 above)
Comment (grosbouddha) [reply src/autoComplete/index.tsx]: Removed (I paramaterized the hasMinCharsForSearch function instead)

File: CHANGELOG.md [hunk 6]
```diff
@@ -10,6 +10,7 @@
 - **[UPDATE]** Removing css linter (legacy, targeting CSS files which does not exist anymore on this repository)
 - **[UPDATE]** Moving to styled-jsx V3
 - **[UPDATE]** Re-organize the dependencies (dev & prod dependencies)
+- **[FIX]** Ensure that autocomplete texfield is updated when its defaultValue props is updated.
```
[the remark this answers is not in the document]
Comment (grosbouddha) [reply CHANGELOG.md]: Done

File: src/autoComplete/index.tsx (hunk 5 above)
Comment (grosbouddha) [reply src/autoComplete/index.tsx]: Done

File: src/autoComplete/index.tsx [hunk 7]
```diff
@@ -56,7 +56,9 @@ interface AutoCompleteProps {
 interface AutoCompleteState {
   readonly busy: boolean
   readonly items: AutocompleteItem[]
-  readonly value: string
+  readonly value: string // Not the textfield value but the formatted value of the selected AC item.
```
[the remark this answers is not in the document]
Comment (grosbouddha) [reply src/autoComplete/index.tsx]: Done

**Pull request merged**